File: Classic_Algorithms/sieve_of_eratosthenese/sieve_of_eratosthenese.py
print("Welcome to the prime number generator, using the Sieve of Eratosthenese algorithm")

num = int(input("Enter the number you want prime numbers calculated up to..."))

num_list = []
not_prime = []

for i in range(1, num + 1):
  num_list.append(i)

for number in num_list:
  if (number % 2 == 0) & (number >= 2*2):
    not_prime.append(number)
  elif (number % 3 == 0) & (number >= 3*3):
    not_prime.append(number)
  elif (number % 5 == 0) & (number >= 5*5):
    not_prime.append(number)

for number in not_prime:
  num_list.remove(number)
  
print(f"Prime numbers: {num_list}")
print(f"Number of prime numbers found: {len(num_list)}")

File: Classic_Algorithms/sieve_of_eratosthenese/sieve_of_eratosthenese.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Please enter the number in which you want prime numbers calculated up to...")
  
  var upto int
  fmt.Scan(&upto)

  num_list := []int{}
  not_prime := []int{}

  for i := 0; i < upto; i++ {
    num_list = append(num_list, i)
  }

  for i := range num_list {
    if (num_list[i] % 2 == 0) && (i >= 2*2) {
      not_prime = append(not_prime, i)
    } else if (i % 3 == 0) && (i >= 3*3) {
      not_prime = append(not_prime, i)
    } else if (i % 5 == 0) && (i >= 5*5) {
      not_prime = append(not_prime, i)
    }
  }

  for i := range not_prime {
    var index int
    for i2 := range num_list {
      if not_prime[i] == num_list[i2] {
        index = i2
      }
    }
    num_list = append(num_list[:index], num_list[index+1:]...)
  }

  fmt.Println("Prime numbers found up to %v", upto)
  fmt.Println(num_list)
}
